src/chunk_index.c

```diff
@@ -261,22 +224,30 @@ chunk_index_insert(int32 chunk_id,
  * relation. This function is typically called when a new chunk is created and
  * it should, for each hypertable index, have a corresponding index of its own.
  */
-static Oid
+static void
 chunk_index_create(int32 hypertable_id,
 				   Relation hypertable_idxrel,
 				   int32 chunk_id,
 				   Relation chunkrel,
-				   bool isconstraint)
+				   Oid constraint_oid)
 {
 	Oid			chunk_indexrelid;
 
-	chunk_indexrelid = chunk_relation_index_create(hypertable_idxrel, chunkrel, isconstraint);
+	if (OidIsValid(constraint_oid))
+	{
+		/*
+		 * If there is an associated constraint then that constraint created
+		 * both the index and the catalog entry for the index
+		 */
+		return;
```

Comment:
Is this something we can really trust ?

Reply:
Again, I think it is a concern that we have two ways to create catalog entries, requiring notes or special handling like this. It seems a lot more logical that catalog entries are created here for all indexes, although we don't recurse the actual index creation for constraint indexes. This avoids separate code for catalog entries in PLPGSQL that is already implemented here. (See other comment related to this).